### PR TITLE
Update the description about DM collation_compatible config

### PR DESCRIPTION
**Pull request by WizardXiao**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
Add DM `collation_compatible ` config description.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/8094
- Other reference link(s): https://github.com/pingcap/tiflow/pull/4121

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: dm/task-configuration-file-full.md [hunk 1]
```diff
@@ -33,6 +33,7 @@ case-sensitive: false           # Determines whether the schema/table is case-se
 online-ddl: true                # Supports automatic processing of upstream "gh-ost" and "pt".
 online-ddl-scheme: "gh-ost"     # `online-ddl-scheme` has been deprecated in v2.0.6, so it is recommended to use `online-ddl`.
 clean-dump-file: true           # Whether to clean up the files generated during data dump. Note that these include `metadata` files.
+collation_compatible: "loose"   # The way to sync default collation in create sql, it can be "loose" or "strict" and "loose" is default. DM will sync default collation from upstream when it is "strict", but not when it is "loose". Note that when it is "strict" and the default collation in upstream doesn't be supportted by downstream, it may occur error.
```
Comment (ran-huang) [inline dm/task-configuration-file-full.md:36]:
```suggestion
collation_compatible: "loose"   # The mode to sync the default collation in `CREATE` SQL statements. The supported values are "loose" (by default) or "strict". When the value is "strict", DM syncs the default collation from the upstream; when the value is "loose", DM does not sync the default collation. In "strict" mode, if the downstream does not support the default collation in the upstream, the downstream may report an error. 
```
@WizardXiao PTAL

Comment (WizardXiao) [reply dm/task-configuration-file-full.md]: yes, that will be much clearer.

**Review by ran-huang — APPROVED** [review]
LGTM

File: dm/task-configuration-file-full.md [hunk 2]
```diff
@@ -33,6 +33,7 @@ case-sensitive: false           # Determines whether the schema/table is case-se
 online-ddl: true                # Supports automatic processing of upstream "gh-ost" and "pt".
 online-ddl-scheme: "gh-ost"     # `online-ddl-scheme` has been deprecated in v2.0.6, so it is recommended to use `online-ddl`.
 clean-dump-file: true           # Whether to clean up the files generated during data dump. Note that these include `metadata` files.
+collation_compatible: "loose"   # The mode to sync the default collation in `CREATE` SQL statements. The supported values are "loose" (by default) or "strict". When the value is "strict", DM syncs the default collation from the upstream; when the value is "loose", DM does not sync the default collation. In "strict" mode, if the downstream does not support the default collation in the upstream, the downstream may report an error.
```
Comment (en-jin19) [inline dm/task-configuration-file-full.md:36]:
```suggestion
collation_compatible: "loose"   # The mode to sync the default collation in `CREATE` SQL statements. The supported values are "loose" (by default) or "strict". When the value is "strict", DM syncs the default collation from the upstream; when the value is "loose", DM does not sync the default collation. In "strict" mode, if the downstream does not support the default collation in the upstream, the downstream might report an error. 
```

File: dm/task-configuration-file-full.md [hunk 3]
```diff
@@ -33,6 +33,7 @@ case-sensitive: false           # Determines whether the schema/table is case-se
 online-ddl: true                # Supports automatic processing of upstream "gh-ost" and "pt".
 online-ddl-scheme: "gh-ost"     # `online-ddl-scheme` has been deprecated in v2.0.6, so it is recommended to use `online-ddl`.
 clean-dump-file: true           # Whether to clean up the files generated during data dump. Note that these include `metadata` files.
+collation_compatible: "loose"   # The mode to sync the default collation in `CREATE` SQL statements. The supported values are "loose" (by default) or "strict". When the value is "strict", DM syncs the default collation from the upstream; when the value is "loose", DM does not sync the default collation. In "strict" mode, if the downstream does not support the default collation in the upstream, the downstream might report an error.
```
Comment (en-jin19) [inline dm/task-configuration-file-full.md:36]:
```suggestion
collation_compatible: "loose"   # The mode to sync the default collation in CREATE SQL statements. The supported values are "loose" (by default) or "strict". When the value is "strict", DM syncs the default collation from the upstream; when the value is "loose", DM does not sync the default collation. In "strict" mode, if the downstream does not support the default collation in the upstream, the downstream might report an error. 
```

**Review by dveeden — APPROVED** [review]

File: dm/task-configuration-file-full.md (hunk 3 above)
Comment (lance6716) [inline dm/task-configuration-file-full.md:36]:
```suggestion
collation_compatible: "loose"   # The mode to sync the default collation in `CREATE` SQL statements. The supported values are "loose" (by default) or "strict". When the value is "strict", DM appends the according collation of the upstream to SQL; when the value is "loose", DM does not modify the SQL. In "strict" mode, if the downstream does not support the default collation in the upstream, the downstream might report an error. 
```

**Review by lance6716 — COMMENTED** [review]
rest lgtm

**Review by lance6716 — APPROVED** [review]

**Pull request merged**